### PR TITLE
Add env support

### LINKED ISSUE #39: Improved "environment" functionality - Tracking and control of loaded js/css libraries.
*Update*: Keeping track of env objects was going to be a pain, so refactored to take a much simpler approach

**Revised approach** 

CAT will display links to all loaded `<link>` and `<script>` tags under the environment section of the CAT controls. Link tags will have a toggle that allows the user to enable/disable links to see how css effects the charts. Users can add any css files they want to the page hosting the CAT instance to facilitate testing of different environments. 

Not going to deal with "unloading" scripts at this time. 

**Original plan**

Add support for new `config.env` option where `config.env` is an array with objects that have the following properties: 

`config.env.label` - a human readable name for the environment
`config.env.js` - an array of paths for javascript to be loaded before rendering the chart
`config.css.js` - an array of paths for css to be loaded before renderering the chart

Each env option will appear in a drop down along with "none".

### PR DESCRIPTION
**Pull request by jwildfire**
closes #39

### CODE REVIEW

**Comment by jwildfire** [comment]
Added @rbudrevich as a 2nd reviewer for funsies. Just the last 3 commits related to #39 need attention in this PR. Note that #46 is a known issue and will get fixed before this is released.

File: index.html [hunk 1]
```diff
@@ -17,6 +17,8 @@
         <script type='text/javascript' src ='https://cdn.rawgit.com/RhoInc/viz-library/7783f26e/util/web/data/dataFiles.js'></script>
 
         <link type = 'text/css' rel = 'stylesheet' href = './css/cat.css'>
+        <link type = 'text/css' rel = 'stylesheet' href = 'https://cdn.rawgit.com/RhoInc/webcharts/master/css/webcharts.css'>
```
Comment (samussiah) [inline index.html]: Load webcharts css by default?

File: css/cat.css [hunk 2]
```diff
@@ -120,3 +120,77 @@
...
+.cat-wrap .environment-section ul{
+  list-style: none;
+  padding-left: 0.1em;
+}
+
+.cat-wrap .environment-section ul h5{
+  margin:0;
+}
+/**********************************************************************
+** Sliders from https://www.w3schools.com/howto/howto_css_switch.asp **
```
Comment (samussiah) [inline css/cat.css]: purdy

File: src/cat/env/showEnv.js [hunk 3]
```diff
@@ -1,20 +1,73 @@
...
+    .attr("class", "switch")
+    .classed("hidden", d => d.filename == "cat.css");
+
+  var switchCheck = switchWrap
+    .append("input")
+    .property("type", "checkbox")
+    .property("checked", d => !d.disabled);
+  switchWrap.append("span").attr("class", "slider round");
+
+  switchCheck.on("click", function(d) {
```
Comment (samussiah) [inline src/cat/env/showEnv.js]: nice

**Comment by samussiah** [comment]
added a polyfill for Array.find() and removed a couple console.log()s

**Pull request merged**